src/utils/S3/index.ts

```diff
@@ -0,0 +1,55 @@
+import AWS from "aws-sdk";
+import { accessKeyId, secretAccessKey, bucketName } from "./config";
+
+const s3: AWS.S3 = new AWS.S3({
+  accessKeyId,
+  secretAccessKey,
+});
+
+const uploadToS3 = async ({
+  image_base64,
+  file_name,
+}): Promise<string> | null => {
+  const params = {
+    Bucket: bucketName,
+    Key: file_name,
+    Body: Buffer.from(
+      image_base64.replace(/^data:image\/\w+;base64,/, ""),
+      "base64"
+    ),
+    ContentEncoding: "base64",
+    ContentType: "image/jpeg",
+  };
+
+  const uploadImage = s3.upload(params).promise();
+  const S3Response = await uploadImage
+    .then(data => data)
+    .catch(err => {
+      console.log(err);
+      return null;
+    });
+
+  if (!S3Response) return null;
+
+  const { Location: image_url } = S3Response;
+  return image_url;
```

Comment:
Personally I'll use `try catch`
```
try {
  const S3Response = await uploadImage()
  ...
  return image_url
} catch (error) {
  console.log(error)
  return null
```

honestly I'm not sure if `S3Response` is a false value in the event of an error during `uploadImage`

Reply:
thas sounds like a good idea, thanks for the insight! i'll try do more research on this thenn